internal/connector/connector.go

```diff
@@ -98,123 +101,79 @@ func (j *jwkCache) getJWK() (*jose.JSONWebKey, error) {
 	return &response.Keys[0], nil
 }
 
-var JWKCacheRefresh = 5 * time.Minute
-
-type BearerTransport struct {
-	Token     string
-	Transport http.RoundTripper
-}
-
-func (b *BearerTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	if b.Token != "" {
-		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", b.Token))
+func newAuthenticator(url string, options Options) *authenticator {
+	// nolint:forcetypeassert // http.DefaultTransport is always http.Transport
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{
+		//nolint:gosec // We may purposely set InsecureSkipVerify via a flag
+		InsecureSkipVerify: options.SkipTLSVerify,
 	}
 
-	return b.Transport.RoundTrip(req)
+	return &authenticator{
+		client:  &http.Client{Transport: transport},
+		baseURL: url,
+	}
 }
 
-type getJWKFunc func() (*jose.JSONWebKey, error)
-
-func jwtMiddleware(getJWK getJWKFunc) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		authorization := c.GetHeader("Authorization")
-
-		raw := strings.ReplaceAll(authorization, "Bearer ", "")
-		if raw == "" {
-			logging.L.Debug("no bearer token found")
-			c.AbortWithStatus(http.StatusUnauthorized)
-
-			return
-		}
-
-		tok, err := jwt.ParseSigned(raw)
-		if err != nil {
-			logging.S.Debugf("invalid jwt signature: %v", err)
-			c.AbortWithStatus(http.StatusUnauthorized)
-
-			return
-		}
-
-		key, err := getJWK()
-		if err != nil {
-			logging.L.Debug("could not get jwk")
-			c.AbortWithStatus(http.StatusUnauthorized)
-
-			return
-		}
-
-		var claims struct {
-			jwt.Claims
-			claims.Custom
-		}
-
-		out := make(map[string]interface{})
-		if err := tok.Claims(key, &claims, &out); err != nil {
```

Comment:
I'm guessing `out` was only used for debugging, I did not see it referenced anywhere else. I removed it for now.